src/components/Menu.vue

```diff
@@ -0,0 +1,105 @@
+<script>import algoliaComponent from '../component';
+  import { FACET_OR } from '../store';
+
+  export default {
+    mixins: [algoliaComponent],
+
+    props: {
+      attribute: {
+        type: String,
+        required: true,
+      },
+      limit: {
+        type: Number,
+        default: 10,
+      },
+      sortBy: {
+        default() {
+          return ['isRefined:desc', 'count:desc', 'name:asc'];
+        },
+      },
+    },
+
+    computed: {
+      facetValues() {
+        return this.searchStore.getFacetValues(
+          this.attribute,
+          this.sortBy,
+          this.limit
+        );
+      },
+    },
+
+    methods: {
+      handleClick(event, value) {
+        event.preventDefault();
+  
+        let alreadyRefined = false;
+        for (const refinement of this.searchStore.activeRefinements) {
+          if (
+            refinement.attributeName === this.attribute &&
+            refinement.name === value
+          ) {
+            alreadyRefined = true;
+            break;
+          }
+        }
```

Comment:
```js
const alreadyRefined = this.searchStore.activeRefinements
  .some(({ attributeName, name }) => attributeName === this.attribute && name === value);
```

Reply:
That looks like Scala ;) go for it!

Reply:
JavaScript nowadays is just Scala but the the tiny bit more verbose we need to allow mortals to understand it